app/resources/device.py

```diff
@@ -55,40 +49,47 @@ def get_all(data: dict, user: str) -> dict:
     """
     devices: List[Device] = wrapper.session.query(Device).filter_by(owner=user).all()
 
-    return {
-        "devices": [d.serialize for d in devices]
-    }
+    return {"devices": [d.serialize for d in devices]}
 
 
-@m.user_endpoint(path=["device", "create"], requires={})
+@m.user_endpoint(path=["device", "create"], requires=requirement_build)
 def create(data: dict, user: str) -> dict:
     """
     Create a device.
     :param data: The given data.
     :param user: The user uuid.
     :return: The response
     """
-    device_count = wrapper.session.query(Device).filter_by(owner=user).first()
+    comp, message = check_compatible(data)
+    if not comp:
+        return message
+
+    comp, message = check_exists(user, data)
+    if not comp:
+        return message
+
+    performance: tuple = calculate_power(data)
+
+    device: Device = Device.create(user, True)
+
+    Workload.create(device.uuid, performance)
 
-    if device_count:
-        return already_own_a_device
+    create_hardware(data, device.uuid)
 
-    device: Device = Device.create(user, 1, True)
+    delete(user, data)
```

Comment:
Rename this function. This would call the handler of the delete endpoint

Reply:
renamed done